service/controller/v24/adapter/cpf_route_tables.go

```diff
@@ -2,109 +2,118 @@ package adapter
 
 import (
 	"context"
-	"fmt"
-	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/aws-operator/service/controller/v24/controllercontext"
 	"github.com/giantswarm/aws-operator/service/controller/v24/encrypter"
-	"github.com/giantswarm/aws-operator/service/controller/v24/key"
+	"github.com/giantswarm/aws-operator/service/routetable"
 )
 
+type CPFRouteTablesConfig struct {
+	RouteTable *routetable.RouteTable
+
+	AvailabilityZones []v1alpha1.AWSConfigStatusAWSAvailabilityZone
+	EncrypterBackend  string
+	NetworkCIDR       string
+}
+
 type CPFRouteTables struct {
+	routeTable *routetable.RouteTable
+
 	PrivateRoutes []CPFRouteTablesRoute
 	PublicRoutes  []CPFRouteTablesRoute
+
+	availabilityZones []v1alpha1.AWSConfigStatusAWSAvailabilityZone
+	encrypterBackend  string
+	networkCIDR       string
+}
+
+func NewCPFRouteTables(config CPFRouteTablesConfig) (*CPFRouteTables, error) {
```

Comment:
I don't know. The type is public and it has to be public I guess because of the go templating. Having a private constructor for a public type is weird. I think we should aim for dropping the constructors because effectively we should simply inject information and not have any logic in the template rendering. It will take some time but this would be my goal.

Reply:
I have zero issues with having private constructor for public type. It has clear notion that the type is meant to be used outside the package but it's constructed internally.

Reply:
done